source/lac/trilinos_vector_base.cc

```diff
@@ -372,7 +372,7 @@ namespace TrilinosWrappers
     // is zero on _all_ processors.
     const Epetra_MpiComm *mpi_comm
       = dynamic_cast<const Epetra_MpiComm *>(&vector->Map().Comm());
-    Assert(mpi_comm != 0, ExcInternalError());
+    AssertThrow(mpi_comm != 0, ExcInternalError());
```

Comment:
Why did you change all the `Assert` to `AssertThrow`. We have the convention that internal errors (such as this one) should be governed by `Assert`, not `AssertThrow`. (By the way, both can throw an exception - the first one is meant for internal error conditions, the latter is meant for error conditions due to user input/interaction that should always be thrown).